File: src/util/time.rs
use chrono::{DateTime, Utc};

/// Format a timestamp as a relative time string like "2h", "3d", "1w"
pub fn relative_time(dt: &DateTime<Utc>) -> String {
    let now = Utc::now();
    let duration = now.signed_duration_since(*dt);

    let minutes = duration.num_minutes();
    let hours = duration.num_hours();
    let days = duration.num_days();
    let weeks = days / 7;

    if minutes < 1 {
        "now".to_string()
    } else if minutes < 60 {
        format!("{}m", minutes)
    } else if hours < 24 {
        format!("{}h", hours)
    } else if days < 7 {
        format!("{}d", days)
    } else if weeks < 52 {
        format!("{}w", weeks)
    } else {
        format!("{}y", days / 365)
    }
}

pub fn is_stale(dt: &DateTime<Utc>, threshold_hours: u64) -> bool {
    let now = Utc::now();
    let duration = now.signed_duration_since(*dt);
    duration.num_hours() as u64 >= threshold_hours
}

File: src/lib/time.ts
export function relativeTime(isoDate: string): string {
  const now = Date.now();
  const then = new Date(isoDate).getTime();
  const diffMs = now - then;
  const minutes = Math.floor(diffMs / 60_000);
  const hours = Math.floor(diffMs / 3_600_000);
  const days = Math.floor(diffMs / 86_400_000);
  const weeks = Math.floor(days / 7);

  if (minutes < 1) return "now";
  if (minutes < 60) return `${minutes}m`;
  if (hours < 24) return `${hours}h`;
  if (days < 7) return `${days}d`;
  if (weeks < 52) return `${weeks}w`;
  return `${Math.floor(days / 365)}y`;
}

export function isStale(isoDate: string, thresholdHours: number): boolean {
  const now = Date.now();
  const then = new Date(isoDate).getTime();
  const hours = (now - then) / 3_600_000;
  return hours >= thresholdHours;
}
